libraries/AP_Generator/AP_Generator_IE_2400.h

```diff
@@ -0,0 +1,60 @@
+#pragma once
+
+#include "AP_Generator_IE_FuelCell.h"
+
+#ifndef GENERATOR_ENABLED
+#define GENERATOR_ENABLED !HAL_MINIMIZE_FEATURES
+#endif
+
+#if GENERATOR_ENABLED
+
+class AP_Generator_IE_2400 : public AP_Generator_IE_FuelCell
+{
+    // Inherit constructor
+    using AP_Generator_IE_FuelCell::AP_Generator_IE_FuelCell;
+
+public:
+    void update(void) override;
+
+
+private:
+    // Process characters recieved and build sentence for IE 2.4kW
+    void decode_latest_term(void) override;
+
+    // Convert error code from fuel cell to AP failsafe bitmask
+    void set_failsafe_mask(uint32_t err_in);
+
+    // Check error codes and populate message with error code
+    bool check_err_code(char* msg_txt, uint8_t msg_len, bool not_changed) override;
+
+    void Log_Write(void);
+
+    // IE 2.4kW failsafes and corresponding bitmask lookup table
+    enum class ERR_CODE_IE24 {
+        MINOR_INTERNAL = 1, // Minor internal error is to be ignored
+        REDUCED_POWER = 10,
+        SPM_LOST = 11,
+        PRESSURE_LOW = 20,
+        BATTERY_LOW = 21,
+        PRESSURE_ALERT = 30,
+        START_DENIED = 31,
+        SYSTEM_CRITICAL = 32,
+        PRESSURE_CRITICAL = 33,
+        BATTERY_CRITICAL = 40,
+    };
+
+    struct Lookup_FS_IE24 {
+        ERR_CODE_IE24 option;
+        const uint32_t bit_mask;
+    };
+
+    static const Lookup_FS_IE24 lookup_fs_ie24[];
+
+    // Data that is received over telem and is only being logged
```

Comment:
Not sure you need this structure here.  Also, last_time_ms is not received over telemetry :-)

Reply:
So, I am a little confused as to when it is appropriate to add a structure in AP.  I appreciate that only one instance is ever created here, but it acts a a good collector for the variables to easily see where they are used.  You asked me to add a structure for a similar purpose for 'parsed variables' so I am now a little confused.  Can you you give me some guidance please.